modules/storage/filesystem/device.go

```diff
@@ -18,8 +17,9 @@ type DeviceManager interface {
 	Devices(ctx context.Context) ([]Device, error)
 	// ByLabel finds all devices with the specified label
 	ByLabel(ctx context.Context, label string) ([]Device, error)
-	// PoolType finds the type of a storagepool
-	PoolType(ctx context.Context, pool Pool) (DeviceType, error)
+	// Scan the system for devices. This must be called after all actions which
```

Comment:
> This must be called after all actions which make persistent changes on the disk layout.

Why do you have to scan every disk any time you do something on them ?
This doesn't solve the problem. The goal was to avoid to have to read all the disk every time to know their types. 
Seems now you keep that info in memory. But then now we have a scan function that needs to be called every time? This solution still doesn't allow us to bring the unused disk down.

Reply:
We can avoid this by manually setting labels on disks/partitions which get included in filesystems, setting the device children if we partition it, ...

Reply:
@zaibon https://github.com/threefoldtech/zosv2/issues/84